src/modules/cbm/CbmTypes.h

```diff
@@ -62,25 +124,73 @@ class CbmType {
     } else if (type_int == 0x06) {
       beamMonitorType = CbmTypes::MM;
     } else {
-      throw std::invalid_argument("Invalid CBM type integer: " + std::to_string(type_int));
+      throw std::invalid_argument("Invalid CBM type integer: " +
+                                  std::to_string(type_int));
+    }
+  }
+
+  /**
+   * @brief Converts the CBM type to its string representation.
+   * @return The string representation of the CBM type.
+   */
+  const char *to_string() const {
+    switch (beamMonitorType) {
+    case CbmTypes::TTL:
+      return "TTL";
+    case CbmTypes::N2GEM:
+      return "N2GEM";
+    case CbmTypes::IBM:
+      return "IBM";
+    case CbmTypes::GEM:
+      return "GEM";
+    case CbmTypes::FC:
+      return "FC";
+    case CbmTypes::MM:
+      return "MM";
+    default:
+      return "Unknown";
     }
   }
 
 private:
   CbmTypes beamMonitorType;
 
 public:
-    operator int() const {
-        return static_cast<int>(beamMonitorType);
-    }
+  /**
+   * @brief Conversion operator to int.
+   * @return The integer representation of the CBM type.
+   */
+  operator int() const { return static_cast<int>(beamMonitorType); }
 
-    operator CbmTypes() const {
```

Comment:
I don't like operator overload for (). IMO it is fine to do it for == for which the intention is clear
but () is confusing.

We have had a really good example for a bad use of too much operator overloading in the past (JADAQ repository for the old multiblade readout system). Please reconsider this.